include/seqan3/io/structure/structure_file_format_dot_bracket.hpp

```diff
@@ -0,0 +1,399 @@
+ *
+ * ### Fields
+ *
+ * The Dot Bracket format provides the fields seqan3::field::SEQ, seqan3::field::ID, seqan3::field::STRUCTURE,
+ * seqan3::field::STRUCTURED_SEQ and seqan3::field::ENERGY.
+ * The first three of these fields are required when writing, except seqan3::field::SEQ and seqan3::field::STRUCTURE
+ * can be replaced by seqan3::field::STRUCTURED_SEQ.\n
+ * If you select seqan3::field::STRUCTURED_SEQ you must not select seqan3::field::SEQ or seqan3::field::STRUCTURE.
+ *
+ * ### Implementation notes
+ *
+ * When reading the ID-line the identifier (either `;` or `>`) and any blank characters before the actual ID are
+ * stripped. Each field is read/written as a single line.
+ *
+ * This implementation supports the following less known and optional features of the format:
+ *
+ *   * ID lines beginning with `;` instead of `>`
+ *   * character counts and spaces within the sequence (they are simply ignored)
+ *
+ * The following optional features are currently **not supported:**
+ *
+ *   * Multiple comment lines (starting with either `;` or `>`), only one ID line before the sequence line is accepted
+ *
+ */
+class structure_file_format_dot_bracket
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \brief Rule of five explicitly defaulted.
+     * \{
+     */
+    structure_file_format_dot_bracket() = default;
+    structure_file_format_dot_bracket(structure_file_format_dot_bracket const &) = delete;
+    structure_file_format_dot_bracket & operator=(structure_file_format_dot_bracket const &) = delete;
+    structure_file_format_dot_bracket(structure_file_format_dot_bracket &&) = default;
+    structure_file_format_dot_bracket & operator=(structure_file_format_dot_bracket &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "dbn" }
+    };
+
+    //!\copydoc structure_file_in_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type,
+              bool     structured_seq_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename bpp_type,
+              typename structure_type,
+              typename energy_type,
+              typename react_type,
+              typename comment_type,
+              typename offset_type>
+    void read(stream_type & stream,
+              structure_file_in_options<seq_legal_alph_type, structured_seq_combined> const & options,
+              seq_type & seq,
+              id_type & id,
+              bpp_type & bpp,
+              structure_type & structure,
+              energy_type & energy,
+              react_type & react,
+              react_type & react_err,
+              comment_type & comment,
+              offset_type & offset)
```

Comment:
You probably get a lot of unused variable warning from this! I would wrap the unused ones in SEQAN3_DOXYGEN_ONLY

Aso you could align the `&` but you don't have to.